bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/firmware/Firmware.java

```diff
@@ -285,11 +299,27 @@ public boolean isPrerequisiteVersion(String firmwareVersion) {
         return new Version(firmwareVersion).compare(internalPrerequisiteVersion) >= 0;
     }
 
+    public boolean isSuitableFor(Thing thing) {
+        return hasSameThingType(thing) && hasRequiredModel(thing);
+    }
+
     @Override
     public int compareTo(Firmware firmware) {
         return -internalVersion.compare(new Version(firmware.getVersion()));
     }
 
+    private boolean hasSameThingType(Thing thing) {
+        return Objects.equal(this.getUID().getThingTypeUID(), thing.getThingTypeUID());
+    }
+
+    private boolean hasRequiredModel(Thing thing) {
+        if (isModelRestricted()) {
+            return Objects.equal(this.getModel(), thing.getProperties().get(PROPERTY_MODEL_ID));
```

Comment:
see above

Reply:
Here, in contrast to above, the two compared objects could both be null. Hence, `Objects.equals()` is, I think, the right choice. (Of course using the `java.util.Objects` class as suggested above.)